### ISSUE TITLE
Problem installing gem

### ISSUE THREAD

**Issue by mchapman** [ged/ruby-pg#309]
Running on Ubuntu 16.04 I am getting the following:
```
$ gem install pg -v 1.1.4
Building native extensions. This could take a while...
ERROR:  Error installing pg:
	ERROR: Failed to build gem native extension.

    current directory: /home/mark/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/pg-1.1.4/ext
/home/mark/.rbenv/versions/2.5.1/bin/ruby -r ./siteconf20191125-26053-aidwby.rb extconf.rb
checking for pg_config... yes
Using config values from /usr/bin/pg_config
checking for libpq-fe.h... *** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
	--with-opt-dir
	--without-opt-dir
	--with-opt-include
	--without-opt-include=${opt-dir}/include
	--with-opt-lib
	--without-opt-lib=${opt-dir}/lib
	--with-make-prog
	--without-make-prog
	--srcdir=.
	--curdir
	--ruby=/home/mark/.rbenv/versions/2.5.1/bin/$(RUBY_BASE_NAME)
	--with-pg
	--without-pg
	--enable-windows-cross
	--disable-windows-cross
	--with-pg-config
	--without-pg-config
	--with-pg_config
	--without-pg_config
	--with-pg-dir
	--without-pg-dir
	--with-pg-include
	--without-pg-include=${pg-dir}/include
	--with-pg-lib
	--without-pg-lib=${pg-dir}/lib
/home/mark/.rbenv/versions/2.5.1/lib/ruby/2.5.0/mkmf.rb:456:in `try_do': The compiler failed to generate an executable file. (RuntimeError)
You have to install development tools first.
	from /home/mark/.rbenv/versions/2.5.1/lib/ruby/2.5.0/mkmf.rb:590:in `try_cpp'
	from /home/mark/.rbenv/versions/2.5.1/lib/ruby/2.5.0/mkmf.rb:1150:in `block in find_header'
	from /home/mark/.rbenv/versions/2.5.1/lib/ruby/2.5.0/mkmf.rb:947:in `block in checking_for'
	from /home/mark/.rbenv/versions/2.5.1/lib/ruby/2.5.0/mkmf.rb:350:in `block (2 levels) in postpone'
	from /home/mark/.rbenv/versions/2.5.1/lib/ruby/2.5.0/mkmf.rb:320:in `open'
	from /home/mark/.rbenv/versions/2.5.1/lib/ruby/2.5.0/mkmf.rb:350:in `block in postpone'
	from /home/mark/.rbenv/versions/2.5.1/lib/ruby/2.5.0/mkmf.rb:320:in `open'
	from /home/mark/.rbenv/versions/2.5.1/lib/ruby/2.5.0/mkmf.rb:346:in `postpone'
	from /home/mark/.rbenv/versions/2.5.1/lib/ruby/2.5.0/mkmf.rb:946:in `checking_for'
	from /home/mark/.rbenv/versions/2.5.1/lib/ruby/2.5.0/mkmf.rb:1149:in `find_header'
	from extconf.rb:54:in `<main>'

To see why this extension failed to compile, please check the mkmf.log which can be found here:

  /home/mark/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/extensions/x86_64-linux/2.5.0/pg-1.1.4/mkmf.log

extconf failed, exit code 1

Gem files will remain installed in /home/mark/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/pg-1.1.4 for inspection.
Results logged to /home/mark/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/extensions/x86_64-linux/2.5.0/pg-1.1.4/gem_make.out
$ cat /home/mark/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/extensions/x86_64-linux/2.5.0/pg-1.1.4/mkmf.log
find_executable: checking for pg_config... -------------------- yes

--------------------

"gcc -o conftest -I/home/mark/.rbenv/versions/2.5.1/include/ruby-2.5.0/x86_64-linux -I/home/mark/.rbenv/versions/2.5.1/include/ruby-2.5.0/ruby/backward -I/home/mark/.rbenv/versions/2.5.1/include/ruby-2.5.0 -I. -I/usr/include/postgresql -I/home/mark/.rbenv/versions/2.5.1/include    -O3 -ggdb3 -Wall -Wextra -Wno-unused-parameter -Wno-parentheses -Wno-long-long -Wno-missing-field-initializers -Wno-tautological-compare -Wno-parentheses-equality -Wno-constant-logical-operand -Wno-self-assign -Wunused-variable -Wimplicit-int -Wpointer-arith -Wwrite-strings -Wdeclaration-after-statement -Wimplicit-function-declaration -Wdeprecated-declarations -Wno-packed-bitfield-compat -Wsuggest-attribute=noreturn -Wsuggest-attribute=format -Wno-maybe-uninitialized  -fPIC conftest.c  -L. -L/home/mark/.rbenv/versions/2.5.1/lib -Wl,-rpath,/home/mark/.rbenv/versions/2.5.1/lib -L/usr/lib/x86_64-linux-gnu -Wl,-rpath,/usr/lib/x86_64-linux-gnu -L. -L/home/mark/.rbenv/versions/2.5.1/lib  -fstack-protector -rdynamic -Wl,-export-dynamic     -Wl,-rpath,/home/mark/.rbenv/versions/2.5.1/lib -L/home/mark/.rbenv/versions/2.5.1/lib -lruby  -lpthread -lgmp -ldl -lcrypt -lm   -lc"
/usr/bin/ld: cannot find -lgmp
collect2: error: ld returned 1 exit status
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: int main(int argc, char **argv)
4: {
5:   return 0;
6: }
/* end */

$ ls -al /usr/include/postgresql
total 124
drwxr-xr-x  4 root root  4096 Nov 25 12:37 .
drwxr-xr-x 49 root root 20480 Nov 25 12:51 ..
drwxr-xr-x  3 root root  4096 Nov 25 12:37 internal
drwxr-xr-x  2 root root  4096 Nov 25 12:37 libpq
-rw-r--r--  1 root root  2214 Nov 12 11:33 libpq-events.h
-rw-r--r--  1 root root 22839 Nov 12 11:33 libpq-fe.h
-rw-r--r--  1 root root   323 Nov 12 11:33 pg_config_ext.h
-rw-r--r--  1 root root 32776 Nov 12 11:33 pg_config.h
-rw-r--r--  1 root root 11687 Nov 12 11:33 pg_config_manual.h
-rw-r--r--  1 root root  1052 Nov 12 11:33 pg_config_os.h
-rw-r--r--  1 root root  2239 Nov 12 11:33 postgres_ext.h
```
I am sure there is something elementary I am missing - can anyone help?  In the absence of the Google Plus groups referred to in the readme is the appropriate place to ask for this kind of help?

**Comment by larskanis** [comment]
The interesting line is
```
/usr/bin/ld: cannot find -lgmp
```
So you probably have to
```
sudo apt install libgmp-dev
```

**Comment by mchilson** [comment]
I am having this same issue trying to install the pg gem on a clean install of Ubuntu 20.04 LTS. Postgres 12 is the default to install on this release and the compiler cannot find a header file... Here is the failing line:

Using config values from /usr/bin/pg_config
checking for libpq-fe.h... no
**Can't find the 'libpq-fe.h header**

I could be wrong but the issue seems to be that one of the dependencies of libpq.dev is causing the issue. 
I did another clean install in a VM and got the same issue. 
A little more experimenting I installed Postgres 10 in a clean VM and everything worked

I have all the correct libraries installed so I know I am good there (this is not my first rodeo with pg and rails).

Thoughts?

Many thanks in advance.
Mike
